src/co_refract.jl

```diff
@@ -0,0 +1,159 @@
+# This file is a part of AstroLib.jl. License is MIT "Expat".
+
+function _co_refract(old_alt::T, altitude::T, pressure::T, temperature::T,
+                    epsilon::T, to_observe::Bool) where {T<:AbstractFloat}
+
+    if isnan(temperature)
+        if altitude > 11000
+            temperature = 211.5
+        else
+            temperature = 283 - 0.0065 * altitude
+        end
+    end
+
+    if isnan(pressure)
+        pressure = 1010 * (1 - 6.5/288000 * altitude) ^ 5.255
+    end
+
+    if !to_observe
+        aout = old_alt - co_refract_forward(old_alt, pressure, temperature)
+    else
+        cur = old_alt + co_refract_forward(old_alt, pressure, temperature)
+        last = zero(T)
+        while true
```

Comment:
I don't think it'll be necessary ;-)  I have to admit that I didn't look to the IDL code before posting the comments, now I have to say that your solution was almost good: in IDL a [`repeat...until`](https://www.harrisgeospatial.com/docs/repeat___until.html) loop is used, which evaluates the body of the loop at least once, much like the [`do...while`](https://en.wikipedia.org/wiki/Do_while_loop) loop in C.  Unfortunately, Julia doesn't have such construct, but you can get the same effect with:
```julia
while true
    last = cur
    cur = old_alt + co_refract_forward(last, pressure, temperature)
    if abs(last-cur) * 3600 < epsilon
        break
    end
end
```
Sorry for not having noticed this earlier.